src/Microsoft.AspNetCore.OData/Batch/ODataBatchHttpRequestMessageExtensions.cs

```diff
@@ -188,12 +191,21 @@ internal static Task CreateODataBatchResponseAsync(this HttpRequest request, IEn
             writerSettings.MessageQuotas = messageQuotas;
 
             HttpResponse response = request.HttpContext.Response;
-            response.StatusCode = (int)HttpStatusCode.OK;
 
-            // Need to get the stream from ODataBatchContent.
-            // maybe use the shared class to define bahaviour and put
-            // HttpContent and public stream getter in platform-specific?
-            ODataBatchContent batchContent = new ODataBatchContent(responses, requestContainer);
+            StringValues acceptHeader = request.Headers["Accept"];
+            string contentType = null;
+            if (StringValues.IsNullOrEmpty(acceptHeader)
+                || acceptHeader.Any( h => h.Equals(ODataBatchHttpRequestExtensions.BatchMediaTypeMime, StringComparison.OrdinalIgnoreCase)))
+            {
+                contentType = String.Format(CultureInfo.InvariantCulture, "multipart/mixed;boundary=batchresponse_{0}", Guid.NewGuid());
```

Comment:
>multipart/mixed [](start = 75, length = 15)

use "BatchMediaTypeMime"

Reply:
Since this is combined with trailing hard-coded string "boundary=...", we can just use hard-coded string for better readibility, which is for the same reason as used in original ODataBatchContent L34.
Does that make sense?

---
In reply to: [202868054](https://github.com/OData/WebApi/pull/1536#discussion_r202868054) [](ancestors = 202868054)